ethreceipts/ethreceipts.go

```diff
@@ -821,19 +826,19 @@ func collectOk[T any](in []T, oks []bool, okCond bool) []T {
 // 	return txnLogs
 // }
 
-func groupLogsByTransaction(logs []types.Log, numTxns int) [][]*types.Log {
-	if numTxns == 0 {
-		return [][]*types.Log{}
-	}
-	out := make([][]*types.Log, blockLogsCount(numTxns, logs))
+func groupLogsByTransaction(logs []types.Log) map[string][]*types.Log {
+	var out = make(map[string][]*types.Log)
 	for _, log := range logs {
 		log := log
-		out[log.TxIndex] = append(out[log.TxIndex], &log)
-	}
-	for i, logs := range out {
-		if logs == nil {
-			out[i] = []*types.Log{}
+
+		logTxHash := log.TxHash.Hex()
+		outLogs, ok := out[logTxHash]
+		if !ok {
+			outLogs = []*types.Log{}
 		}
```

Comment:
This isn't necessary I guess since if it's nil it will automatically be created by the append.

Reply:
Ohh, true, however I will leave it as it is. I have already tagged it.